lib/jekyll/page.rb

```diff
@@ -66,18 +65,10 @@ def dir
       end
     end
 
-    # The full path and filename of the post. Defined in the YAML of the post
-    # body.
-    #
-    # Returns the String permalink or nil if none has been set.
-    def permalink
-      data.nil? ? nil : data['permalink']
-    end
-
     # The template of the permalink.
     #
     # Returns the template String.
-    def template
+    def url_template
```

Comment:
* This is a breaking change, please consider reverting this name change.  
* This change will impact what milestone it can be used at, it breaks the API.